drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -321,13 +321,44 @@ function DragAndDropTemplates(configuration) {
         )
     };
 
-    var mainTemplate = function(ctx) {
-        var problemTitle = ctx.show_title ? h('h2.problem-title', {innerHTML: ctx.title_html}) : null;
-        var problemHeader = ctx.show_problem_header ? h('h3.title1', gettext('Problem')) : null;
+    var itemFeedbackPopupTemplate = function(ctx) {
         var popupSelector = 'div.popup';
-        if (ctx.popup_html && !ctx.last_action_correct) {
+        var msgs = ctx.feedback_messages || [];
+        var have_messages = msgs.length > 0;
+        var popup_content;
+
+        if (msgs.length > 0 && !ctx.last_action_correct) {
             popupSelector += '.popup-incorrect';
         }
+
+        if (ctx.mode == DragAndDropBlock.ASSESSMENT_MODE) {
+            var content_items = [
+                (!ctx.last_action_correct) ? h("p", {}, gettext("Some of your answers were not correct")) : null,
+                h("p", {}, gettext("Hints:")),
+                h("ul", {}, msgs.map(function(message) {
+                    return h("li", {innerHTML: message.message});
+                }))
+            ];
+            popup_content = h("div.popup-content", {}, have_messages ? content_items : []);
+        } else {
+            popup_content = h("div.popup-content", {}, msgs.map(function(message) {
+                return h("p", {innerHTML: message.message});
+            }))
+        }
+
+        return h(
+            popupSelector, {style: {display: have_messages ? 'block' : 'none'} },
+            [
+                h('div.close.icon-remove-sign.fa-times-circle'),
+                popup_content
+            ]
+        )
+    };
+
+    var mainTemplate = function(ctx) {
+        var problemTitle = ctx.show_title ? h('h2.problem-title', {innerHTML: ctx.title_html}) : null;
```

Comment:
This needs to be an h3 (and the one below an h4). There is already an h2 on the page (course header). If you look at other components, you see that their display names are h3 instead of h2.